pkg/aws/metadata/handler_credentials.go

```diff
@@ -56,12 +58,29 @@ func (c *credentialsHandler) Handle(ctx context.Context, w http.ResponseWriter,
 		return http.StatusInternalServerError, fmt.Errorf("error finding pod for ip %s: %s", ip, err.Error())
 	}
 
+	requestedRole := mux.Vars(req)["role"]
+
+	if foundRole == "" && c.defaultRole == requestedRole {
+		metrics.GetOrRegisterMeter("credentialsHandler.defaultRole", metrics.DefaultRegistry).Mark(1)
+		resp, err := http.Get(fmt.Sprintf("http://169.254.169.254/latest/meta-data/iam/security-credentials/%s", c.defaultRole))
+		if err != nil {
+			return resp.StatusCode, err
+		}
+		body, err := ioutil.ReadAll(resp.Body)
```

Comment:
I'd prefer not to have any of the http.Get code but if it was necessary I think it's missing a `defer resp.Body.Close()` before the reader attempts to read.